Python/Product/TestAdapter/visualstudio_py_testlauncher.py

```diff
@@ -177,6 +177,12 @@ def addUnexpectedSuccess(self, test):
         self._setResult(test, 'passed')
 
     def _setResult(self, test, outcome, trace = None):
+        if ("ptvsd" in sys.modules
```

Comment:
This is a very weird place to be adding this. The code in `main()` already sets the entry point and dont debug lists. Did that not work?

Reply:
In the case where a python test script is launched and immediately attached, the code in main is executed. 

If the python test script is run without debugging and the user later attaches to it (legacy TCP or PID), this code will execute which prevents the user from debugging visualstudio_py_testlauncher.py

Reply:
You should be able to do this bit on top level, outside of main. It doesn't block anything, it will work even if ptvsd is not loaded, and the only side effect that it has, you want to always occur if ptvsd *is* loaded, regardless of the reason why it happens.

Reply:
Never mind - if it's on global level (or in main), then the check for ptvsd can indicate that it's not loaded, but then it gets loaded later in various attach scenarios... but since this module is already loaded, the code won't re-run.

So it has to be on some code path that is guaranteed to be re-executed periodically. Putting it here will make it perform the check after every test, but before any other code runs in this file.

I'm not sure this is 100% foolproof, since by the time we get here, this code is already running... so `DONT_DEBUG` will not apply at least to this very frame. I think the user might end up here if they do Step Out, or Step Over on the last line of their test, if they attached during that test.

A 100% foolproof way would be to register an import hook to detect exactly when ptvsd gets loaded. However, this would require a great deal more code, and very convoluted code at that. It's probably not worth the bother for such a niche scenario.

Reply:
We haven't been hiding this code from debugging for this scenario (run + manual attach) since we first allowed debugging when running tests, and no one ever filed an issue for that. I don't see the need to be fixing this now in legacy debugger when we're so close to removing support.

If this was clean code, sure, but this is an ugly hack.